conformance/utils/suite/experimental_suite.go

```diff
@@ -0,0 +1,278 @@
+	// minimum a list of features which they support.
+	if s.SupportedFeatures == nil && s.ConformanceProfiles.Len() == 0 && !s.EnableAllSupportedFeatures {
+		return nil, fmt.Errorf("no conformance profile was selected for test run, and no supported features were provided so no tests could be selected")
+	}
+
+	// test suite callers can potentially just run all tests by saying they
+	// cover all features, if they don't they'll need to have provided a
+	// conformance profile or at least some specific features they support.
+	if s.EnableAllSupportedFeatures {
+		s.SupportedFeatures = AllFeatures
+	} else {
+		if s.SupportedFeatures == nil {
+			s.SupportedFeatures = sets.New[SupportedFeature]()
+		}
+		// the use of a conformance profile implicitly enables any features of
+		// that profile which are supported at a Core level of support.
+		for _, conformanceProfileName := range s.ConformanceProfiles.UnsortedList() {
+			conformanceProfile, err := getConformanceProfileForName(conformanceProfileName)
+			if err != nil {
+				return nil, fmt.Errorf("failed to retrieve conformance profile: %w", err)
+			}
+			for _, f := range conformanceProfile.ExtendedFeatures.UnsortedList() {
+				if s.SupportedFeatures.Has(f) {
+					if suite.extendedSupportedFeatures[conformanceProfileName] == nil {
+						suite.extendedSupportedFeatures[conformanceProfileName] = sets.New[SupportedFeature]()
+					}
+					suite.extendedSupportedFeatures[conformanceProfileName].Insert(f)
+				} else {
+					if suite.extendedUnsupportedFeatures[conformanceProfileName] == nil {
+						suite.extendedUnsupportedFeatures[conformanceProfileName] = sets.New[SupportedFeature]()
+					}
+					suite.extendedUnsupportedFeatures[conformanceProfileName].Insert(f)
+				}
+			}
+		}
+	}
+
+	if s.FS == nil {
+		s.FS = &conformance.Manifests
+	}
+
+	suite.ConformanceTestSuite = ConformanceTestSuite{
+		Client:           s.Client,
+		RoundTripper:     roundTripper,
+		GatewayClassName: s.GatewayClassName,
+		Debug:            s.Debug,
+		Cleanup:          s.CleanupBaseResources,
+		BaseManifests:    s.BaseManifests,
+		Applier: kubernetes.Applier{
+			NamespaceLabels:          s.NamespaceLabels,
+			ValidUniqueListenerPorts: s.ValidUniqueListenerPorts,
+		},
+		SupportedFeatures: s.SupportedFeatures,
+		TimeoutConfig:     s.TimeoutConfig,
+		SkipTests:         sets.New(s.SkipTests...),
+		FS:                *s.FS,
+	}
+
+	// apply defaults
+	if suite.BaseManifests == "" {
+		suite.BaseManifests = "base/manifests.yaml"
+	}
+
+	return suite, nil
+}
+
+// -----------------------------------------------------------------------------
+// Conformance Test Suite - Public Methods
+// -----------------------------------------------------------------------------
+
+// Setup ensures the base resources required for conformance tests are installed
+// in the cluster. It also ensures that all relevant resources are ready.
+func (suite *ExperimentalConformanceTestSuite) Setup(t *testing.T) {
```

Comment:
It would be helpful to understand how this is different from our main `Setup` function: https://github.com/kubernetes-sigs/gateway-api/blob/c67f3fd31976ea74ea06a78c2a88ae38df95a41d/conformance/utils/suite/suite.go#L147.

Reply:
It's just a copy that allows the experimental version. When this is no longer experimental we will merge the two suites.

Reply:
That makes sense, thanks! Adding some kind of comment in the code to describe this would be helpful for people that aren't as familiar with this plan.

Reply:
I added some comments here 76dcc19